client/bin/multi-client.js

```diff
@@ -474,24 +480,29 @@ class State {
 		while (runQueue.length > 0) {
 			const job = runQueue.pop();
 			// If matching workers for the job are available then allot them a job
-			if (job.matchingDevices !== null) { // specifically for an application since worker URL's are specific are automatically allocated
-				for (var device of job.matchingDevices) {
-					// check if device is idle & public URL is reachable
-					let deviceUrl = await balenaCloud.resolveDeviceUrl(device)
-					try{
-						let status = await rp.get(`${deviceUrl}/state`);
-						if (status === "IDLE") {
-							// make sure that the worker being targetted isn't already about to be used by another child process
-							if(!busyWorkers.includes(deviceUrl)){
-								// Create single client and break from loop to job the job 👍
-								job.workers = deviceUrl
+			for (var device of job.matchingDevices) {
+				// check if device is idle & public URL is reachable
+				let deviceUrl = ''
+				if(!job.array){
+					deviceUrl = await balenaCloud.resolveDeviceUrl(device)
+				} else {
+					deviceUrl = device;
+				}
+				try{
+					let status = await rp.get(`${deviceUrl}/state`);
```

Comment:
```suggestion
					let status = await rp.get(new url.URL('/state', deviceUrl));

```

to fix https://github.com/balena-os/leviathan/pull/537/files#r752156464

Reply:
ah, yes, as above :)